src/Collections-Unordered/Dictionary.class.st

```diff
@@ -320,16 +320,11 @@ Dictionary >> at: key ifPresent: aPresentBlock ifAbsentPut: anAbsentBlock [
 ]
 
 { #category : #accessing }
-Dictionary >> at: key put: anObject [ 
+Dictionary >> at: key put: anObject [
 	"Set the value at key to be anObject.  If key is not found, create a
 	new entry for key and set is value to anObject. Answer anObject."
 
-	| index assoc |
-	index := self findElementOrNil: key.
-	assoc := array at: index.
-	assoc
-		ifNil: [self atNewIndex: index put: (Association key: key value: anObject)]
-		ifNotNil: [assoc value: anObject].
+	self add: (Association key: key value: anObject).
```

Comment:
Out of curiosity, why not using :
```suggestion
	self add: key -> anObject.
```

So there is no direct dependency on the Association class?

Reply:
#-> does reference Association. So it will not change anything. But another issue it that -> is made to be used by the user library.